main.tf

```diff
@@ -3,61 +3,52 @@ resource "aws_cognito_user_pool" "this" {
 
   sms_authentication_message = var.auth_sms_message
   email_verification_message = var.html_email_message == "" ? var.auth_sms_message : var.html_email_message
-  email_verification_subject = "Código de verificación"
-  mfa_configuration          = "OPTIONAL"
+  email_verification_subject = var.email_verification_subject
 
-  admin_create_user_config {
-    allow_admin_create_user_only = false
-  }
-
-  software_token_mfa_configuration {
-    enabled = true
-  }
-
-  auto_verified_attributes = ["email"]
-
-  device_configuration {
-    challenge_required_on_new_device      = true
-    device_only_remembered_on_user_prompt = false
-  }
+  auto_verified_attributes = var.auto_verified_attributes
```

Comment:
Este atributo no tiene valor por defecto en la definicion de la variable, debe ser un parametro opcional

Reply:
I added the auto_verified_attributes variable with a default value of ["email", "phone_number"] in variables.tf to make it optional. I also updated main.tf to use this variable.